src/scaladoc/scala/tools/nsc/doc/html/HtmlFactory.scala

```diff
@@ -115,7 +118,32 @@ class HtmlFactory(val universe: doc.Universe, val reporter: Reporter) {
       finally out.close()
     }
 
+    def copyWebjarResource(resourceName: String, expectedSRI: String): Unit = {
+      import java.security.MessageDigest
+      import java.util.Base64
+      val md = MessageDigest.getInstance("SHA-256")
+      val base64encoder = Base64.getEncoder
+
+      // https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity
+      def calsSubResourceIntegrity(input: String): String = {
+        val messageDigest = md.digest(input.getBytes)
+        val base64encoded = base64encoder.encode(messageDigest)
+        new String(base64encoded, "UTF-8")
+      }
+      val fileContent = Source.fromResource(resourceName).mkString
+      val actualSRI = calsSubResourceIntegrity(fileContent)
+      if (actualSRI != expectedSRI)
+        throw new Exception(s"Subresource Integrity unmatched: expected ${expectedSRI} but ${actualSRI}")
```

Comment:
is it worth saying what file SRI failed for?

Reply:
True.
Added resource name instead of SRI value (SRI value does not help when invalid resource found)